### DISCUSSION TITLE
Export types in nuxt module

### DISCUSSION THREAD

**Discussion by beyer-johannes** [nuxt/nuxt#16885] [Questions]
I did not deeply understood the nuxt modules export logic and how does it work if a nuxt module is used as npm package. 
I'm happy you can help me with following:

As you can see in my example here: https://github.com/beyer-johannes/module-export-example I have an enum "CarTypes" 

```
// /packages/example-module/src/runtime/carTypes.ts

export enum CarTypes {
  capriolet = 'CAPRIOLET',
  limousine = 'LIMOUSINE',
  van = 'VAN',
}
```
which I want to export to make it reusable from any other nuxt project like

```
// /packages/example-project/server/api/cars.get.ts
import { CarTypes } from '@example/example-module';

export default defineEventHandler(async (event) => {
  return {
    data: [
      {
        title: 'Car 1',
        type: CarTypes.capriolet,
      },
      {
        title: 'Car 2',
        type: CarTypes.capriolet,
      },
      {
        title: 'Car 3',
        type: CarTypes.van,
      },
    ],
  };
});
```
## Question

How can I reach, that in my modules dist directory is my CarTypes enum is exported as well.

## Note
- I want to use a monorepo with pnpm workspaces
- I not want to auto import it with addAutoImport from nuxt kit

Thanks for any advice!

**Comment by stenet** [comment]
@beyer-johannes have you already found a solution for this?

**Reply by beyer-johannes** [reply]
@stenet no - I posted this multiple times on discord but got no answer. May @danielroe can you help us here please?

**Reply by stenet** [reply]
see [https://github.com/nuxt/module-builder/issues/39](https://github.com/nuxt/module-builder/issues/39)

**Comment by beyer-johannes** [comment]
@danielroe @pi0 can anyone please help us here. I can not proceed my work, because I stuck here since one month :/
